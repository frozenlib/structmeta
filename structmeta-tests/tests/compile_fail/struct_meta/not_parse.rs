#[derive(structmeta::StructMeta)]
struct Example {
    not_parse: NotParse,
}
struct NotParse;

fn main() {}
