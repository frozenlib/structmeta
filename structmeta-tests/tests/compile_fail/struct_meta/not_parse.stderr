error[E0277]: the trait bound `NotParse: Parse` is not satisfied
 --> tests/compile_fail/struct_meta/not_parse.rs:3:16
  |
3 |     not_parse: NotParse,
  |     ---------  ^^^^^^^^ the trait `Parse` is not implemented for `NotParse`
  |     |
  |     required by a bound introduced by this call
  |
  = help: the following other types implement trait `Parse`:
            Example
            std::boxed::Box<T>
            TokenStream
            TokenTree
            proc_macro2::Group
            proc_macro2::Punct
            proc_macro2::Ident
            proc_macro2::Literal
          and $N others
note: required by a bound in `ParseBuffer::<'a>::parse`
 --> $CARGO/syn-2.0.29/src/parse.rs
  |
  |     pub fn parse<T: Parse>(&self) -> Result<T> {
  |                     ^^^^^ required by this bound in `ParseBuffer::<'a>::parse`
